homeassistant/components/savant/coordinator.py

```diff
@@ -0,0 +1,86 @@
+"""Provides DataUpdateCoordinator."""
+
+from datetime import timedelta
+import logging
+import typing
+
+from pysavant.switch import AudioSwitch, Switch, VideoSwitch
+
+from homeassistant.components.button import ButtonEntity
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.exceptions import ConfigEntryError
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .media_player import SavantPlayer
+
+logger = logging.getLogger(__name__)
+
+
+class SavantCoordinator(DataUpdateCoordinator):
+    """DataUpdateCoordinator for Savant matrix switch."""
+
+    players: list[SavantPlayer] = []
+    sensors: list[SensorEntity] = []
+    buttons: list[ButtonEntity] = []
+    info: dict[str, typing.Any]
+    api: Switch | None = None
+
+    def __init__(self, hass, entry):
+        """Initialize my coordinator."""
+        super().__init__(
+            hass,
+            logger,
+            name=entry.data["name"],
+            config_entry=entry,
+            update_interval=timedelta(seconds=1),
+            always_update=True,
+        )
+        match entry.data["type"]:
+            case "Audio":
+                self.api = AudioSwitch(entry.data["ip"])
+            case "Video":
+                self.api = VideoSwitch(entry.data["ip"])
+            case _:
+                raise ConfigEntryError
```

Comment:
So why are these in here? How do they differ? Can't we inject them instead? Why do we need 2 separate coordinators that don't do anything more than extending?